### PR TITLE
Make use of the full range of AUX matrices

### PR DESCRIPTION
**Pull request by gdonval**
The gist of it is that we have this signature `mr2d_malloc(Int n)`, bounding the internal `malloc` to `Int`, which is Scalapack's indexing type. This is not how `malloc` is defined and means that maximum AUX matrix size is artifically limited on 64-bit systems to 2GB.

This PR operates this transformation: `mr2d_malloc(Int n) -> mr2d_malloc(size_t n)`, yet ensures `Int -> size_t` conversion does not involve Int-negative values and ensures that no 64-bit values are passed to `malloc` on 32-bit systems.

The main advantage of this on 64-bit system allows the use of the **full** _signed_ 32-bit indexing range instead of `range / element size`. E.g. the max AUX matrix size is now 16GB instead of 2GB previously.

With this, "standard" 32-bit Scalapack and Blas/Lapack can still be used in programs like GPAW.

Details (and full commit history) are in
https://github.com/Reference-ScaLAPACK/scalapack/pull/85 which does not seem to receive much attention.

This patch is a way to provide that feature to conda users in the meantime.

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] ~~Reset the build number to `0` (if the version changed)~~
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by gdonval** [comment]
@conda-forge-admin, please rerender

**Comment by gdonval** [comment]
Those osx failures are unrelated to my code (different file, different folder, error completely unrelated to anything I have done).

**Comment by gdonval** [comment]
Alright, I took the easy approach to solving the osx problems: ignore those specific warnings.

1. Those warnings are also shown with Linux and can only be solved with a new Scalapack release.
2. Those warnings are ignored on Linux already (and one of them is already ignored in the existing recipe for osx-arm specifically).
3. The builds are tested on Linux and Osx so a problem here should show up there.

In short, this is ready for release.

**Comment by Chronum94** [comment]
Any news/updates on this? Or is this specific feedstock abandoned?

**Comment by gdonval** [comment]
Well. still waiting for review... and since I'm not added as a maintainer yet, nothing I can do.

**Comment by dalcinl** [comment]
@gdonval I'm not in a position to review this PR. But if you ask me explicitly to merge and you take accountability for the outcome, I'm very happy to click the button.

**Comment by dalcinl** [comment]
@conda-forge-admin rerender

**Comment by gdonval** [comment]
@dalcinl: I'll recheck tomorrow to make sure all looks good and I'll ping you if so. Thanks a lot!

**Comment by traversaro** [comment]
> In short, this is ready for release.

In the future, please add a mention to the mantainer team (i.e. @conda-forge/scalapack) when you write that a PR is ready for review. Otherwise, it is quite easy for mantainers to miss a message.

File: recipe/build.sh [hunk 1]
```diff
@@ -1,5 +1,14 @@
 #!/bin/sh
 
+if [[ "$target_platform" == osx-* ]]
+then
+  # Workaround for implicit function declarations on osx 
+  export CFLAGS="$CFLAGS -std=gnu99 -Wno-implicit-function-declaration"
+  export CXXFLAGS="$CXXFLAGS -std=gnu99 -Wno-implicit-function-declaration"
+  export FFLAGS="${FFLAGS} -fallow-argument-mismatch"
+  export OMPI_FCFLAGS=${FFLAGS}
```
Comment (traversaro) [inline recipe/build.sh]: Can you explain why this is necessary?

File: recipe/build.sh [hunk 2]
```diff
@@ -8,7 +17,7 @@ if [[ "$target_platform" == "osx-64" ]]; then
   fi
 fi
 
-if [[ "$target_platform" == linux-* || "$target_platform" == "osx-arm64"  ]]
```
Comment (traversaro) [inline recipe/build.sh]: I guess the workaround is not needed anymore in osx-arm64 due to an update in the compiler, at this point probably we can also remove the comment in the if referring to gfortran 9?

**Comment by Chronum94** [comment]
@gdonval did you get a chance to check things? Just a ping since this will improve a few downstream things.

**Comment by traversaro** [comment]
@gdonval @Chronum94 are you still interested in this? @Chronum94 even if you were not the original author of the PR, if the original author does not reply you can open a new PR with your own improvements/addressing the reviews.

**Comment by minrk** [comment]
Rebased to just the patch commit, since this no longer needs any build.sh patches. Thanks @gdonval!

**Comment by gdonval** [comment]
Sorry for the delay. Happy this got merged! 🎉
